### PR TITLE
Support binary data download

### LINKED ISSUE #103: Binary data download
Currently perron is converting the response body into a string assuming that it's a valid UTF-8 string. This prevents users from using perron for fetching binary data.

https://github.com/zalando-incubator/perron/blob/f880daedf83835191bc12c5bab034ce2a9aaa7c8/lib/request.ts#L324

### PR DESCRIPTION
**Pull request by spark59**
Fixes #103

### CODE REVIEW

File: test/request.test.js [hunk 1]
```diff
@@ -130,6 +130,20 @@ describe("request", () => {
...
+    const promise = request({
+      autoDecodeUtf8: false
+    });
+    const responseStub = new ResponseStub();
+    requestStub.emit("response", responseStub);
+    responseStub.emit("data", Buffer.from("foo"));
+    responseStub.emit("data", Buffer.from("bar"));
+    responseStub.emit("end");
+    return promise.then(response => {
+      assert.equal(true, response.body instanceof Buffer);
```
Comment (shuhei) [inline test/request.test.js:143]:
```suggestion
      assert.deepEqual(response.body, Buffer.from("foobar"));
```

**Comment by shuhei** [comment]
We made PRs almost at the same time somehow 😄 

Could you also document this option in README?

**Comment by spark59** [comment]
Will do 👍 Thank you!

File: test/request.test.js (hunk 1 above)
Comment (spark59) [reply test/request.test.js]: 👍

File: lib/request.ts [hunk 2]
```diff
@@ -321,7 +323,14 @@ export const request = (
           bufferLength
         );
         hasRequestEnded = true;
-        const body = Buffer.concat(chunks, bufferLength).toString("utf8");
+
+        let body;
+        const bufferedBody: Buffer = Buffer.concat(chunks, bufferLength);
```
Comment (ruiaraujo) [inline lib/request.ts:328]: I have noticed the type of body in the `ServiceClientResponse` is any, maybe we could type it as `Buffer | string | object`.

Bonus points if we can type it conditionally based on the options. :smile:

Comment (spark59) [reply lib/request.ts:328]: I think it would be more convenient for clients if we decode based on the `content-type` in the response

**Comment by shuhei** [comment]
👍

**Comment by spark59** [comment]
👍

**Comment by spark59** [comment]
@ruiaraujo Could you help get this PR merged?  :)

**Comment by shuhei** [comment]
Published as v0.11.4

**Pull request merged**